tests/test_context.py

```diff
@@ -77,3 +82,120 @@ def test_failed_dependency_description(self, mocker):
 ```"""
             in description
         )
+
+    def test_setup_llm_client_no_env_vars(self, mocker):
```

Comment:
could add some `caplog` assertions for the logging, maybe that would clear up my misunderstanding above